### PR TITLE
Update tests for Path changes

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Updates for https://github.com/dotnet/coreclr/pull/16478

Tagged these tests with #27269 as they're related. Will update for both PRs from CoreCLR.

cc: @danmosemsft

### CODE REVIEW

File: src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/AssemblyCatalogTests.cs [hunk 1]
```diff
@@ -109,6 +109,14 @@ public static void Constructor_InvalidFileNameAsCodeBaseArgument_ShouldThrowArgu
             });
         }
 
+        public static void Constructor_InvalidFileNameAsCodeBaseArgument_ShouldThrowIO(Func<string, AssemblyCatalog> catalogCreator)
+        {
+            Assert.ThrowsAny<IOException>(() =>
+            {
+                var catalog = catalogCreator("??||>");
+            });
```
Comment (stephentoub) [inline src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/AssemblyCatalogTests.cs]: Nit: doesn't need to store the result:
```C#
Assert.ThrowsAny<IOException>(() => catalogCreator("??||>"));
```

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 2]
```diff
@@ -34,14 +34,21 @@ public void EmptyAsPath_ThrowsArgumentException()
...
+            Assert.Throws<ArgumentException>(() => Create(invalidPath));
+        }
+
+        [ActiveIssue(27269)]
+        [Theory, MemberData(nameof(PathsWithInvalidCharacters))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void PathWithInvalidCharactersAsPath_Core(string invalidPath)
+        {
+            if (invalidPath.Contains('\0'))
                 Assert.Throws<ArgumentException>(() => Create(invalidPath));
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Is there a param name we can include/test here?

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 3]
```diff
@@ -203,11 +210,29 @@ public void DirectoryWithComponentLongerThanMaxComponentAsPath_ThrowsException(s
...
-        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Versions of netfx older than 4.6.2 throw an ArgumentException instead of NotSupportedException. Until all of our machines run netfx against the actual latest version, these will fail.")]
-        public void PathWithInvalidColons_ThrowsNotSupportedException(string invalidPath)
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+        public void PathWithInvalidColons_ThrowsNotSupportedException_Desktop(string invalidPath)
         {
-            Assert.Throws<NotSupportedException>(() => Create(invalidPath));
+            if (PathFeatures.IsUsingLegacyPathNormalization())
+            {
+                Assert.Throws<ArgumentException>(() => Create(invalidPath));
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Same question for all of these arg exceptions.

File: src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs [hunk 4]
```diff
@@ -140,14 +140,27 @@ public void AllowedSymbols()
...
 
+        [ActiveIssue(27269)]
+        [Theory,
+            MemberData(nameof(ControlWhiteSpace))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void WindowsControlWhiteSpace_Core(string component)
+        {
+            // CreateSubdirectory will throw when passed a path with control whitespace e.g. "\t"
+            string path = IOServices.RemoveTrailingSlash(GetTestFileName());
```
Comment (maryamariyan) [inline src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs]: I'm wondering why we have this path statement here in the existing test you copied when it's not going to throw exception? the input argument is ```component``` which is being used in the next statement not to compute path. 
Am I missing something?

File: src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/AssemblyCatalogTests.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/AssemblyCatalogTests.cs]: I was following along with the rest of the file's style. I'll go ahead and change all of them.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Not for desktop, but now that I've split it up I can (and will) add it for core.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 3 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: For desktop the answer is no for ArgumentException. IOException doesn't have a parameter and I'm not sure we could add one in a way that made sense.

File: src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs (hunk 4 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs]: I just missed the fact that it wasn't used as I plowed through these. Good catch.

**Pull request merged**